src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -1201,5 +1203,69 @@ uint ITypesDebugInfoWriter.GetMemberFunctionId(MemberFunctionIdTypeDescriptor me
         {
             return GetMemberFunctionIdTypeIndex(_nativeObjectWriter, memberIdDescriptor);
         }
+
+        private static string GetLLVMTripleFromTarget(TargetDetails target)
+        {
+            // We create a triple based on the Target
+            // Detect the LLVM arch
+            string arch = null;
+            // Not used
+            string sub = string.Empty;
+            switch (target.Architecture)
+            {
+                case TargetArchitecture.ARM:
+                    arch = "thumbv7";
+                    break;
+                case TargetArchitecture.ARM64:
+                    arch = "aarch64";
+                    break;
+                case TargetArchitecture.X64:
+                    arch = "x86_64";
+                    break;
+                case TargetArchitecture.X86:
+                    arch = "x86";
+                    break;
+                case TargetArchitecture.Wasm32:
+                    arch = "wasm32";
+                    break;
+                default:
+                    throw new InvalidOperationException($"The architecture `{target.Architecture}` is not supported by ObjectWriter");
+            }
+
+            string vendor = null;
+            string sys = null;
+            string abi = null;
+            switch (target.OperatingSystem)
+            {
+                case TargetOS.Windows:
+                    vendor = "pc";
+                    sys = "win32";
+                    abi = "windows";
+                    break;
+                case TargetOS.Linux:
+                case TargetOS.FreeBSD:
+                case TargetOS.NetBSD:
+                    // TODO: vendor for ARM/ARM64 could be changed? We might need another information in TargetDetails to make a decision
```

Comment:
Does the vendor actually affects how the object writer works? I doubt it. I would worry about this only once somebody hits it as a real problem.